### ISSUE TITLE
Non Combat Move Unloading From Transports Unloaded During Combat Move (2.0.18289)

### ISSUE THREAD

**Issue by Cernelius** [triplea-game/triplea#6039]
So, this is a problem I already reported, and it got closed without a fix, so opening it again, with hope.

CORRECT BEHAVIOUR: In "World War II Revised" (thus in any games following the same ruleset, like "Big World", "Pact of Steel", "New World Order", "World At War", and so on), as well as in "Europe" and "Pacific" (the originals), owned or allied units that offloaded one or more units, during the Combat Move phase, can still offload units, during the Non Combat Move phase, but all units offloaded, during the Non Combat Move phase, from a same one of these units, if any, may only do so together into the same zone as the units that were offloaded, from the same unit, during the Combat Move phase.

CURRENT WRONG BEHAVIOUR: In "World War II Revised", as well as in "Europe" and "Pacific" (the originals), units that offloaded any units, during Combat Move, cannot offload any units, during Non Combat Move.

OTHER INFORMATION: This behaviour applies only to "V2" rules games, except not to "V2LHTR" rules, and it also applies to "Europe" and "Pacific", but not to any other versions (it doesn't apply to V1, V3, V4, V5, V6, nor Europe/Pacific/Global 1940 I and II).

Reproduction steps:
Select "World War II Revised" and start the game.
Turn Edit Mode on.
Edit 1 transport and 1 infantry into SZ 4.
Turn Edit Mode off.
Skip to the Combat Move phase.
Load 1 infantry or 1 armour onto the transport in SZ 4 (so having 2 infantries or 1 infantry and 1 armour on the transport), move the transport to SZ 3, offload 1 infantry or 1 armour into Norway, also attacking Norway with everything else at reach, except leaving 1 infantry in the transport.
If you take Norway, verify that the program doesn't let you offload the remaining infantry, during Non Combat Move.

What should have happened, instead:
You should be able to offload the remaining infantry, during Non Combat Move, if you take Norway.

Main user case:
I believe the main user case is having 1 infantry and 1 aaGun already on board of the transport, using the infantry to make a sea borne invasion, then offloading the aaGun, into the same territory, during Non Combat Move. This is a legal sequence of moves, for Revised (except LHTR), that the program doesn't let you do.

If a developer has any will of taking over this matter, I can add more considerations and commentary about how to implement this change (with additional properties, and so on). Sadly, there is the problem that "V2LHTR" lacks a distinct property. However, even making this change for "V2" in general would be good, as "LHTR" is vastly in the minority, both as a number of games and as popularity of them.

@panther2 If you can please confirm the rules at this issue, do so (I don't need it, as I'm sure that all I've said is correct, but maybe not any developers, that might be interested in fixing the problem, are) (also let's not start talking again that from V2LHTR onwards this is always impossible, as I believe I made it clear enough, this time).

**Comment by Cernelius** [comment]
This is the previous issue, reporting the same problem, also having a save game for 1.9.0.0, to test it:
https://github.com/triplea-game/triplea/issues/1148

**Comment by DanVanAtta** [comment]
@Cernelius a save game with the repro steps already done would be really great. Essentially that save would be a case where you can say "this infantry, i can move them to this territory, but they should only be allowed to move this other territory".

Thank you for re-cutting this issue, it seems relatively clear (maybe not perfectly minimized :grin: ,but certainly straight forward and specific enough we know what needs to be done).

Just to be sure, the 'offload to the same territory in non-combat as during combat move', that rule applies to all rule variants?

**Comment by Cernelius** [comment]
>a save game with the repro steps already done would be really great.

Taken with the current stable or with the referenced prerelease?

But I don't think a save is really needed, here. You can just take "Revised" and edit 1 transport with 2 infantry inside. Then, you should be able to offload 1 during CM and the second one during NCM, to the same territory as before. This is easier than downloading and finding a savegame.

>Just to be sure, the 'offload to the same territory in non-combat as during combat move', that rule applies to all rule variants?

No.
I already answered in "OTHER INFORMATION":
>This behaviour applies only to "V2" rules games, except not to "V2LHTR" rules, and it also applies to "Europe" and "Pacific", but not to any other versions (it doesn't apply to V1, V3, V4, V5, V6, nor Europe/Pacific/Global 1940 I and II).

In practice, there was not this rule in the original v1.
This ability, to split offloading between CM and NCM, has been introduced in v2, but taken out in v2LHTR, and never subsequently reintroduced.
However, also "Europe" and "Pacific" have this ability (I mean the originals, called simply as that, that you can find in "Experimental", not the 1940 series of Europe/Pacific/Global), but this is not really important, as those games are both unfinished.

**Comment by DanVanAtta** [comment]
Re: save game

Current latest prerelease. Partially the idea is the save game is attached to this save. During development we typically need to create such save games, it's faster than redoing the edit to see if we fixed the issue exactly. I'd guess we have to recreate a scenario in a game about a half dozen times while we work on fixes to be sure we have everything covered. I'll even sometime take save games if all I need is just to be in phase 3 of the first players turn, it's still faster than clicking done three times.

Out of context, but as a larger scope, I'm thinking it would be really formal if we had a save game attached to each bug that put the game in the exact spot of the bug. It makes it clearer what the bug is, gives a repro case for developers to work with, saves the work of having to create a save game when development takes place and attempts to confirm a fix.

Re: variants

Ah, sorry, I missed that detail. I'm thinking someone will want example maps for each rule variant listed to be sure we are looking at the right example. Rather than documenting this list in just this issue, I created a starter wiki page to define that mapping. Hence forward when we refer to V2LHTR, we'll have a specific reference map for that rule set. @Cernelius would you mind filling the wiki page with some example maps that implement each rule set? https://github.com/triplea-game/triplea/wiki/Game-Rule-Sets

**Comment by Cernelius** [comment]
@Cernelius would you mind filling the wiki page with some example maps that implement each rule set? https://github.com/triplea-game/triplea/wiki/Game-Rule-Sets

I cannot do that the way the header is worded. It should not say "Example Map(s)", it should say "Example Games". In particular, the original V2 and V2LHTR are a same map.

**Comment by Cernelius** [comment]
Also "Global 1940" is not really a ruleset. Even if we ignore all the various "alpha", it should be at least split into "I" and "II" edition (some major rules differences, like the AA guns). The other matter is that v1 also has two different rulesets, the original (second edition, since TripleA starts from that, not from the actual first edition), and the third edition (adding a series of rules changes, some of the implemented, like defensive submersible subs, some of them not, like paratroopers or rolling all dice at once), but now we are clearly going off topic here... Moreover, to minimize confusion, I would say "Europe/Pacific/Global 1940", instead of just "Global 1940".

**Comment by DanVanAtta** [comment]
Feel free to change the table as needed

**Comment by Cernelius** [comment]
>Feel free to change the table as needed

@DanVanAtta Done.
https://github.com/triplea-game/triplea/wiki/Game-Rule-Sets

@panther2 Maybe you want to take a look.

**Comment by Cernelius** [comment]
So, to repeat, of all that list, this problem is relevant for:
V2 Basic
Europe
Pacific

while there is no problem for:
V1.II
V1.III
V2 LHTR
V3
V4
V5 (and V6)
Europe/Pacific/Global 1940.I
Europe/Pacific/Global 1940.II

Meaning that, in all the games at the first list, a same transport can offload part of its cargo in CM and part in NCM, while all the games at the second list never can.

**Comment by Cernelius** [comment]
Also, are you sure you want to call the first column "Rule Variant". Itself, that's fine, but I'd call it "Ruleset", because "variant" is traditionally used to mean a map or game modification, around here. Also, in practice, each ruleset is indeed a "set" of properties, thus it makes the most sense that way, rather than "variant".

**Comment by panther2** [comment]
@Cernelius 
Looks good to me, and I agree with "Ruleset" being more appropriate than "Variant".

**Comment by VictoryFirst2** [comment]
I think the easiest way to solve this is by adding a new game property, and adding that property to the games in question (V2 standard, Europe, Pacific). I suggest calling it:

_Transport Can Unload in Multiple Phases_

@Cernelius To be clear, can the units be unloaded to different territories in each of these games? (I don't think that really matters as that is handled by Transport Restricted Unload, but just want to be sure).

**Comment by Cernelius** [comment]
@VictoryFirst2 Thanks for addressing this issue! I'm looking forward to close this as fixed. It matters not only for the basic games but also for my "270BC Wars" game (as well as "New World Order", "World At War" and so on: you name it).

Please don't forget to have this new property being true as default if the property itself is missing from the game file but there is the `WW2V2` property set true. Unfortunately, this would mean that V2 LHTR games would need to be updated adding this property set as false in order to have the correct behaviour (but, even if they are not, this change would fix many more games that it degrades and allow them all to be fixed).

I don't like the name of the property which you have chosen, but I'll have to think some more about what would be my proposal.

>can the units be unloaded to different territories in each of these games? (I don't think that really matters as that is handled by Transport Restricted Unload, but just want to be sure).

You need to ask @panther2 or someone else than me for an official answer, but I'm certain that in V1 you can offload to unlimited territories from the same sea zone only as a non-combat move and never as a combat move, whereas from V2 onwards you can never offload to more than one territory from the same sea unit on the same turn.

Regarding on whether, in a hypothetical game having `Transport Restricted Unload` false and this new property true (so a hypothetical game in which you can offload into multiple territories during Non-Combat Movement like in V1 but you can also offload during Non-Combat Movement from a ship which offloaded during Combat Movement like in V2.0), the behaviour should be that you can offload into a hostile land zone during combat move and then (without moving the ship) offload into a friendly land zone during non-combat move with a unit you kept on board or you cannot, this is not covered by the rules. Based on the rules, I would say that it would make the most sense that you cannot do so. I don't think anyone can give you a certain answer here: you'll have to decide yourself or the repository owners should do so maybe after asking the community in general or some specific users like @panther2 in particular. Alternatively, I guess you can just do what we know and let the program decide the matter for us by accepting whatever behaviour happens to be for this particular case after the fix for this issue is done and tested successfully.

**Comment by VictoryFirst2** [comment]
> Please don't forget to have this new property being true as default if the property itself is missing from the game file but there is the WW2V2 property set true. Unfortunately, this would mean that V2 LHTR games would need to be updated adding this property set as false in order to have the correct behaviour (but, even if they are not, this change would fix many more games that it degrades and allow them all to be fixed).

Right, but I am not sure if the WW2V2 will override it when the new property is set to false for V2 LHTR. Normally these overarching properties override the settings. I think in that case, it's better to not make it be automatically set by WW2V2, and simply require it to be set manually for each game (as there are only 5 in total or so, I don't think that is a big deal).

**Comment by Cernelius** [comment]
> > Please don't forget to have this new property being true as default if the property itself is missing from the game file but there is the WW2V2 property set true. Unfortunately, this would mean that V2 LHTR games would need to be updated adding this property set as false in order to have the correct behaviour (but, even if they are not, this change would fix many more games that it degrades and allow them all to be fixed).
> 
> Right, but I am not sure if the WW2V2 will override it when the new property is set to false for V2 LHTR. Normally these overarching properties override the settings. I think in that case, it's better to not make it be automatically set by WW2V2, and simply require it to be set manually for each game (as there are only 5 in total or so, I don't think that is a big deal).

I'm as certain as I can be that would not be better. That way you would exchange the current problem with a new problem, namely that the WW2V2 property is not fully changing the game's behaviour from V1 to V2 as it should (as far as I understand, but I guess only @veqryn could confirm this to be true if he remembers that the WW2V2 property is indeed meant fully to switch the entire rule-set from V1 to V2 (like in the case that is the only property you have), and I'm guessing chances are that he does not remember (but would be great if he does and answers)). This said, I'm almost certain that the specific property prevails over the WW2V2 property as a general rule, but it would be better to make sure of that (meaning making sure that, if WW2V2 is true and this property is false, this property does count as false, overriding/ignoring the WW2V2 property) and clearly documenting that this is the way it is supposed to work in PoS2.

I'll drop a link to the related matter (mostly for @veqryn to interpreter the meaning of (I guess his) "ww2v2 rules override many individual rules to create revised rules" comment) even though it is currently closed and waiting for summarisation in a new issue (kind of the same I did in this issue as well):
https://github.com/triplea-game/triplea/issues/5394

(All this said, I'd take a fix for this issue with the (in my opinion) problem of not working with the WW2V2 property over no fix at all.)

**Comment by VictoryFirst2** [comment]
@Cernelius Then the fix would be to make sure WW2V2 does not override settings. Personally, I would like that as I have had troubles in the past with things not working correctly because of isFactory, isStrategicBomber and others being set, overriding other stuff without me knowing. The other option would be to accept the fact that WW2V2 LHTR cannot be an exception for this rule and should use it (which I believe based on what you have said you are also okay with).

To be honest, now I am also quite curious what WW2V2 all sets because there is a reasonable chance it does not set all rules that are supposed to belong to V2. The problem is that it is a painstaking task as you have to search the codebase for WW2V2 and infer from the context what it is doing. It's a shame it is not documented anywhere.

**Comment by Cernelius** [comment]
> [@Cernelius](https://github.com/Cernelius) Then the fix would be to make sure WW2V2 does not override settings.

Based on what I remember of my testing, it does not, but it has been years. I guess I can test it myself with something like 2 properties which I know that they are V2.0 but not V1 (by having them false while WW2V2 is true) and see if the behaviour is indeed that they are false, but the best is obviously that a developer looks into the code of the program and confirms that in every case in which something is set both by the WW2V2 property and one or more other properties, the WW2V2 is ignored if it is true but at least one other property meant to set the same is false.

Let's please not start talking too much about the WW2V2 property: just what's needed for this specific issue.

**Comment by Cernelius** [comment]
> To be honest, now I am also quite curious what WW2V2 all sets because there is a reasonable chance it does not set all rules that are supposed to belong to V2.

Yes, it's not perfect, but let's not turn this issue into an issue for the WW2V2 property problems (which is not).

**Comment by Cernelius** [comment]
> > [@Cernelius](https://github.com/Cernelius) Then the fix would be to make sure WW2V2 does not override settings.
> 
> Based on what I remember of my testing, it does not, but it has been years. I guess I can test it myself with something like 2 properties which I know that they are V2.0 but not V1 (by having them false while WW2V2 is true) and see if the behaviour is indeed that they are false,

Actually, "Pact of Steel" (but not "Pact of Steel 2") can be easily used for testing here.

The behaviour of being shot upon flying over is one which you have in V2 non-LHTR but not in V1 (if @panther2 is as kind as to confirm that what I'm saying about the rules is correct under the understanding that V1 is Classic Second Edition and V2 is Revised). Therefore, if what I said is correct, the property "WW2V2" should have this behaviour to apply if "true" and not to apply if "false" (or absent) inasmuch as the property "Always on AA" is not present in the game file (XML).

"Pact of Steel" does have the "WW2V2" property "true" (non settable by the user).

1) At preset Game Options (click on Reset before starting to be sure), you can start the game and (with way-points), during "Russians Non Combat Move", have the Russian fighter in Karelia S.S.R. move though Western Europe and land on United Kingdom getting AA fired under the fly-over rules (not really correctly implemented, but this is beside the point).

2) Restarting the game with "Always on AA" unticked (meaning "false") and doing the same, you will not get AA fired moving through Western Europe.

3) Then, if you open the game file ("pact_of_steel_2.xml") with a text editor and remove the entire
```
    <property name="Always on AA" value="true" editable="true">
      <boolean/>
    </property>
```
then save and restart the game (making sure that there is no longer the "Always on AA" in Game Options), you will have the same behaviour as if "Always on AA" is present and false.

Since in V2 basic AA should be always on, this either means that

1) (contrary to what I previously said) the WW2V2 option is not actually meant to turn the rule-set from V1 to V2 (but likely only to turn some behaviours from V1 to V2, meaning changing only part of the rule-set), or

2) the WW2V2 option is meant to turn the rule-set from V1 to V2, and the fact that it does not do so for the "Always on AA" property is simply a wrong behaviour (a bug), or

3) the WW2V2 option is meant to turn the rule-set from V1 to V2 only for whatever is generally true for V2, meaning true for both V2 basic and V2 LHTR, so this option is unaffected because it is "true" in basic but false in "LHTR".

My assumption is that point "2" is true.

At this point, either you find something somewhere clarifying whether or not the "WW2V2" property is meant fully to change the rule-set from V1 to V2 or @veqryn actually answers one way or the other or the current TripleA owners (which I guess in theory still comprise @veqryn too?) will have to decide what the "WW2V2" property is supposed to do.

What I mean is that, if you look into what properties the property "WW2V2" sets as "true" and you find that it turns true most of those changing the rule-set from V1 to V2 except only the "Always on AA" one and a few others, I think we can assume that the few ones missing are actually missing (so should be added), otherwise I guess that the "WW2V2" property is actually just a marginal property for setting stuff not otherwise handled.

My assumption remains that the "WW2V2" property's intended behaviour if of both directly applying a number of behaviour and setting true a bunch of properties only if they are not present in the game file with the general intention of turning the entire game from V1 to V2.0 rules with a single property.

I actually have a number of reasons why I think that the "WW2V2" property is meant to do that, but I REALLY don't want to doom this issue too into starting discussing over what the hell the "WW2V2" property is supposed to do...

The conclusion is that I guess it is better to proceed with this issue just by having the behaviour fixed via a property and, once this is done, opening a new issue for determining whether or not the "WW2V2" property should interact in any way with this new property. Of course, this would also mean that only the games getting this new property added to their game file will be actually fixed as part of closing this issue successfully (so most likely very few of them). (If I'm wrong to assume that the "WW2V2" property is meant to turn the rule-set from V1 to V2, then this issue would be truly fully closed.)

Once this issue is fixed, I will consider updating "World At War" as to have this behaviour, but I will preferably wait until the matter regarding its interaction with the "WW2V2" property is clarified (which may make updating unnecessary for this and many other games).

@VictoryFirst2 By now, this issue has become so long that practically either you fix it or I will have to close it and ressumarize it once again for any chance of an other developer fixing it 5 years from now or so.

(After so much time spent on this issue, I DO hope that this is going to get fixed. LOL)

**Comment by Cernelius** [comment]
Once a new property for fixing this issue is added, it will be necessary to open a new issue to determine (or to decide if it cannot be determined) whether or not the "WW2V2" property should set this property "true" if absent. If that is negative, it will be necessary at least to add the new property as "true" (and I suggest non-editable) to the "Pact of Steel" (but not "Pact of Steel 2") and "World War II Revised" (but not "World War II Revised LHTR") games.

**Comment by Cernelius** [comment]
"New World Order", "The Rising Sun" and "World At War" (plus "WAW 1940" too) would also need this new property added to their game files if "WW2V2" is not setting it "true".

**Comment by Cernelius** [comment]
> I think the easiest way to solve this is by adding a new game property, and adding that property to the games in question (V2 standard, Europe, Pacific). I suggest calling it:
> 
> _Transport Can Unload in Multiple Phases_
> 
> [@Cernelius](https://github.com/Cernelius) To be clear, can the units be unloaded to different territories in each of these games? (I don't think that really matters as that is handled by Transport Restricted Unload, but just want to be sure).

@VictoryFirst2 The most correct name I can think of for this property would be

"Units Can Non-Combat Offload From Sea Transports From Which Units Have Offloaded Into Combat".

However, this name is disgustingly long, so my suggestion is actually calling the property

"Sea Transports Can Offload Again After Combat".

If we want to make the property even shorter, it would be fine also to write

"Transports Can Offload Again After Combat" or

"Ships Can Offload Again After Combat".

This last alternative is shorter but fails immediately to clarify that we are talking of sea transports, not of sea units in general.

However, we already have properties like "Surviving Air Move To Land" not bothering to write "Aircraft" or "Air Units" instead of just "Air" as well as properties like "Transport Control Sea Zone" (not "Sea Transports", just "Transport"), properties having "Subs" or even "Sub" instead of "Submarines" and also a property like "Units Can Load In Hostile Sea Zones" ("Units" instead of "Ships" or "Sea Transports").

My conclusion is that my suggested name for this property is

"**Sea Transports Can Offload Again After Combat**".

---

Your suggestion is not bad, by the way. The only problems which I have are that it should be "Offload" instead of "Unload", that phases is too technical in my opinion (and we don't have any properties defining anything in terms of phases), and of course that "Transports" alone could mean that we are talking of land or air transports. Moreover, "multiple" is commonly understood as more than two (and we are usually (actually always in every rule-set which matters) talking about offloading twice per turn in this case)(If I say that there are multiple issues, that does not normally mean that there are a couple issues. If I ask for a multiplayer in the lobby, that does not normally mean that I'm open to play a 1V1.), though it can also be understood as two or more of course. If wanting to go with something like you are proposing, I would rather go with "Sea Transports Can Offload More Than Once Per Turn" or "Sea Transports Can Offload Multiple Times Per Turn".

**Comment by Cernelius** [comment]
I'm going to update the title too.

**Comment by Cernelius** [comment]
@panther2 May be a particularly good person for wording this property well enough based on how things are called.

**Comment by VictoryFirst2** [comment]
@Cernelius Thanks! So to sum up, you'd want a new property called:

**"Sea Transports Can Offload Again After Combat".**

that if set to true, allows a sea transport (not air/land, or naval warship) to offload units in the NCM-phase after already having offloaded units in the CM-phase. This can be to a different territory if "Transport Restricted Unload" is false. If that property is true, then the territories must be the same.

I'd make one small change though, and that is to replace "offload" with "unload". In POS2, the general word is "unload". There is the existing property "Transport Restricted Unload". There is no mention of "offload" at all. So to stay consistent, I'd call it **"Sea Transports Can Unload Again After Combat".**

**Comment by VictoryFirst2** [comment]
@Cernelius 
> @VictoryFirst2 By now, this issue has become so long that practically either you fix it or I will have to close it and ressumarize it once again for any chance of an other developer fixing it 5 years from now or so.

I have been thinking a lot about this and I think the ticket deserves to be reopened. Not so much to document the behavior, but rather to change the way these properties are handled in the code. Right now, they are completely integrated. At various points, you have the if-condition "if property X OR WW2V2 is true, then...". This means that either the relevant property sets the behavior, OR WW2V2. This is the reason why WW2V2 might seem to override the properties. I don't think this is a good way to do this. I think WW2V2 should rather set those properties to true in a separate java file. This allows you to see what it does, and also allows you to make a small change to the WW2V2 ruleset because the WW2V2 property will not override all property settings (unless set at the bottom of the XML of course). This probably needs some discussion first as it would be a major code change.

**Comment by Cernelius** [comment]
> [@Cernelius](https://github.com/Cernelius)
> 
> > [@VictoryFirst2](https://github.com/VictoryFirst2) By now, this issue has become so long that practically either you fix it or I will have to close it and ressumarize it once again for any chance of an other developer fixing it 5 years from now or so.
> 
> I have been thinking a lot about this and I think the ticket deserves to be reopened. Not so much to document the behavior, but rather to change the way these properties are handled in the code. Right now, they are completely integrated. At various points, you have the if-condition "if property X OR WW2V2 is true, then...". This means that either the relevant property sets the behavior, OR WW2V2. This is the reason why WW2V2 might seem to override the properties. I don't think this is a good way to do this. I think WW2V2 should rather set those properties to true in a separate java file. This allows you to see what it does, and also allows you to make a small change to the WW2V2 ruleset because the WW2V2 property will not override all property settings (unless set at the bottom of the XML of course). This probably needs some discussion first as it would be a major code change.

Thanks for having a deep dive into the WW2V2 property, but isn't this largely or even entirely off-topic? This issue is not for the WW2V2 property, and my last take can be summarised in "let's just forget about the WW2V2 property and get this fix done with a stand-alone property". If I would summarise this issue, I would almost certainly not even mention the bloody WW2V2 property at all to avoid it derailing the focus from the actual issue once again. Did you mean to post this in
https://github.com/triplea-game/triplea/issues/5394
?

If so, may you please delete this post and repost it in that issue. I think here we can all agree with forgetting completely about the WW2V2 property until this issue is (hopefully successfully) closed.

What remains to be done in this issue is adding a new property which gives the wanted behaviour if "true", the property being entirely stand-alone.

**Comment by Cernelius** [comment]
@VictoryFirst2 :
> [@Cernelius](https://github.com/Cernelius) Thanks! So to sum up, you'd want a new property called:
> 
> **"Sea Transports Can Offload Again After Combat".**
> 
> that if set to true, allows a sea transport (not air/land, or naval warship) to offload units in the NCM-phase after already having offloaded units in the CM-phase. This can be to a different territory if "Transport Restricted Unload" is false. If that property is true, then the territories must be the same.

Actually, it can be a "naval warship". For example, I want this behaviour for every ship of my "270BC Wars" game, each of which is both a transport and a warship, and the cruiser of "World At War" (being both a transport and a warship) may have this behaviour too (so long as the game gets updated by adding the new property). Also in Classic and Revised transports ARE warships, just with offence/defence values of 0/1.

About the interactions with the "Transport Restricted Unload" property, I've already answered and not changed my mind, but substantially I'm very uncertain about it, so feel free to have it work however you think is better. Either this property being false allowing to offload to an other sea zone during Non-Combat or not is fine with me, even though I tend to think (but very uncertainly so) that you should never be able to do that.

This is what I said:
@Cernelius :
>Regarding on whether, in a hypothetical game having `Transport Restricted Unload` false and this new property true (so a hypothetical game in which you can offload into multiple territories during Non-Combat Movement like in V1 but you can also offload during Non-Combat Movement from a ship which offloaded during Combat Movement like in V2.0), the behaviour should be that you can offload into a hostile land zone during combat move and then (without moving the ship) offload into a friendly land zone during non-combat move with a unit you kept on board or you cannot, this is not covered by the rules. Based on the rules, I would say that it would make the most sense that you cannot do so. I don't think anyone can give you a certain answer here: you'll have to decide yourself or the repository owners should do so maybe after asking the community in general or some specific users like @panther2 in particular. Alternatively, I guess you can just do what we know and let the program decide the matter for us by accepting whatever behaviour happens to be for this particular case after the fix for this issue is done and tested successfully.

@asvitkine :
> I'd make one small change though, and that is to replace "offload" with "unload". In POS2, the general word is "unload". There is the existing property "Transport Restricted Unload". There is no mention of "offload" at all. So to stay consistent, I'd call it **"Sea Transports Can Unload Again After Combat".**

I didn't think about this. I agree. Offload is arguably a better term but not so much so as to break consistency if unload is generally, consistently and even always used for such cases. Better not to confuse people into thinking there is any difference between unloading and offloading. 👍
